### PR TITLE
Forcing attribute_internals.h to use uint64_t's for shift operations.

### PR DESCRIPTION
**Pull request by timothy-king**

### CODE REVIEW

**Comment by aniemetz** [comment]
Were you actually able to reproduce the problem? Because I'm still not.

**Comment by timothy-king** [comment]
> Were you actually able to reproduce the problem? Because I'm still not.
I have not managed to reproduce the cmake with the autotools build.

File: test/unit/expr/attribute_white.h [hunk 1]
```diff
@@ -120,6 +122,18 @@ class AttributeWhite : public CxxTest::TestSuite {
...
+    //
+    // This tests an implicit global property of libcvc4. It is *brittle*. If
+    // this breaks after you deleted an Attribute<Test*, bool> from somewhere in
+    // CVC4, just keep testing more TestFlagXs until this is true for max_id.
+    const uint64_t max_id = std::max({TestFlag1::s_id,
+                                      TestFlag2::s_id,
+                                      TestFlag3::s_id,
+                                      TestFlag4::s_id,
+                                      TestFlag5::s_id});
+    TS_ASSERT_LESS_THAN(32, max_id);
```
[the remark this answers is not in the document]
Comment (timothy-king) [reply test/unit/expr/attribute_white.h]: What it was supposed to do was to ensure that at least 1 one the test flags is >= 32. This is true in practice. But we don't assume an order to when the libraries are loaded. (If we could guarantee libcvc4 was loaded before this library, this should be ok.) I'll drop this test.

**Comment by aniemetz** [comment]
I can run it with uint64_t. What benchmark set would you propose? (All of SMT-LIB is too much.)

**Comment by timothy-king** [comment]
> What benchmark set would you propose? (All of SMT-LIB is too much.)
It is fairly a generic. So more breadth the better. My guess would be quantifiers? Is there is a representative down sampling of SMT-LIB for performance evaluations still?

**Comment by aniemetz** [comment]
I ran it on (lazy) QF_BV yesterday, the difference is not much (37429 solved for `uint32_t d_bit` vs 37422 for `uint64_t d_bit`), almost all were close to time limit:
```
$ compare-smt-runs.py --diff runs/2018-08-27-attributes-shift-32/ runs/2018-08-27-attributes-shift-64/
                    DIRECTORY |    2018-08-27-attributes-shift-32 |    2018-08-27-attributes-shift-64 |                    
                    BENCHMARK |  STAT  RES EXIT  CPU[s] SPACE[MB] |  STAT  RES EXIT  CPU[s] SPACE[MB] |
 log-slicing-bvashr_0595.smt2 |    ok   20    0 1140.36     376.5 |    to    -    2 1200.07     305.2 |
 log-slicing-bvlshr_0870.smt2 |    to    -    2 1200.08     225.9 |    ok   20    0 1168.55     257.2 |
 log-slicing-bvlshr_0964.smt2 |    to    -    2 1200.06     251.2 |    ok   20    0  1114.3     277.6 |
  log-slicing-bvshl_0771.smt2 |    to    -    2 1200.02     235.7 |    ok   20    0  667.34     238.2 |
 log-slicing-bvult_16967.smt2 |    to    -    2 1200.03     500.7 |    ok   20    0 1148.04     509.1 |
       Sage2-bench_10718.smt2 |    mo    -    3 1139.91    8196.6 |    to    -    2 1200.29    7719.5 |
       Sage2-bench_11198.smt2 |    mo    -    3 1053.96    8197.3 |    to    -    2 1200.28    7280.5 |
       Sage2-bench_12398.smt2 |    to    -    2 1200.28    7669.2 |    mo    -    3  1004.8    8196.2 |
       Sage2-bench_13197.smt2 |    ok   10    0  930.65    2183.9 |    to    -    2 1200.12    2132.6 |
       Sage2-bench_13396.smt2 |    to    -    2 1200.29    7833.9 |    mo    -    3 1186.47    8196.7 |
       Sage2-bench_13465.smt2 |    ok   10    0 1178.04    6994.6 |    to    -    2 1200.26    6773.7 |
       Sage2-bench_13870.smt2 |    to    -    2 1200.22    7611.3 |    mo    -    3 1052.08    8196.2 |
        Sage2-bench_1480.smt2 |    mo    -    3 1066.49    8197.0 |    to    -    2 1200.26    6986.4 |
       Sage2-bench_15653.smt2 |    ok   20    0 1132.48     798.9 |    to    -    2 1200.07     768.2 |
        Sage2-bench_1567.smt2 |    ok   10    0 1172.94    2264.6 |    to    -    2 1200.04    1960.3 |
       Sage2-bench_15895.smt2 |    to    -    2 1200.02    2129.1 |    ok   20    0 1170.89    2436.9 |
        Sage2-bench_1614.smt2 |    ok   10    0  904.62    2566.8 |    to    -    2 1200.18    2232.9 |
       Sage2-bench_16383.smt2 |    mo    -    3 1180.55    8196.5 |    to    -    2 1200.26    7300.4 |
       Sage2-bench_17764.smt2 |    to    -    2 1200.28    7377.5 |    mo    -    3 1198.59    8196.2 |
       Sage2-bench_17884.smt2 |    ok   20    0 1148.67     877.8 |    to    -    2 1200.06     865.6 |
        Sage2-bench_2096.smt2 |    to    -    2 1200.15    2193.1 |    ok   10    0 1111.67    2729.2 |
        Sage2-bench_2097.smt2 |    mo    -    3 1161.05    8196.8 |    to    -    2 1200.27    7371.4 |
          Sage2-bench_25.smt2 |    ok   20    0  993.58    3043.7 |    to    -    2  1200.1    2981.6 |
        Sage2-bench_2613.smt2 |    ok   20    0 1137.66    2816.2 |    to    -    2 1200.16    2784.5 |
         Sage2-bench_313.smt2 |    ok   20    0 1175.78    3350.1 |    to    -    2 1200.14    2879.8 |
        Sage2-bench_3163.smt2 |    ok   10    0 1185.47    2367.0 |    to    -    2 1200.01    2264.1 |
        Sage2-bench_4705.smt2 |    ok   10    0 1199.02    4207.5 |    to    -    2 1200.18    3719.3 |
        Sage2-bench_5882.smt2 |    ok   20    0 1122.38    1726.1 |    to    -    2 1200.01    1453.9 |
         Sage2-bench_688.smt2 |    to    -    2 1200.01    2084.9 |    ok   10    0  946.62    2184.3 |
        Sage2-bench_7537.smt2 |    ok   20    0 1187.06     784.0 |    to    -    2 1200.03     764.7 |
        Sage2-bench_7801.smt2 |    ok   20    0 1172.91     783.8 |    to    -    2 1200.06     770.6 |
        Sage2-bench_7915.smt2 |    to    -    2 1200.31    6659.3 |    mo    -    3 1024.18    8197.2 |
        Sage2-bench_7995.smt2 |    to    -    2 1200.07     868.4 |    ok   20    0 1176.19     878.3 |
        Sage2-bench_8101.smt2 |    to    -    2 1200.07     782.6 |    ok   20    0  966.34     784.4 |
         Sage2-bench_873.smt2 |    ok   20    0 1194.51     648.8 |    to    -    2 1200.03     638.2 |
        Sage2-bench_8779.smt2 |    to    -    2  1200.1     767.1 |    ok   20    0  969.52     784.2 |
        Sage2-bench_8853.smt2 |    to    -    2 1200.03     783.0 |    ok   20    0 1104.88     785.1 |
        Sage2-bench_8918.smt2 |    ok   20    0 1185.74     784.4 |    to    -    2 1200.04     765.3 |
        Sage2-bench_9051.smt2 |    mo    -    3   902.5    8196.5 |    to    -    2 1200.28    8167.0 |
         Sage2-bench_915.smt2 |    ok   20    0 1121.18    3178.8 |    to    -    2 1200.18    2747.1 |

```
We don't have a representative down-sampling right now, cluster is also pretty much blocked right now, so any more runs will take a while. Let me know if I should run other logics (and which).

**Comment by timothy-king** [comment]
Looks good. Can we go forward with the current version and submit the uint64_t version later?

**Comment by aniemetz** [comment]
If uint64_t looks good to you I'd rather have it included here (since it should not require too many changes, right?). Alternatively, open an issue.

**Pull request merged**